app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt

```diff
@@ -255,13 +255,13 @@ class ThreadViewModel @Inject constructor(
             }
         }
 
-        threadState.shouldMarkThreadAsSeen = thread.unseenMessagesCount > 0
+        if (threadState.shouldMarkThreadAsSeen == null) threadState.shouldMarkThreadAsSeen = thread.unseenMessagesCount > 0
```

Comment:
Can't the boolean be non-nullable and the condition be computed right next to where it's used?